pkg/api/node_api.go

```diff
@@ -458,20 +457,52 @@ func (a *NodeApi) stateHash(w http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
+func wavesAddressInvalidCharErr(invalidChar rune, id string) *apiErrs.CustomValidationError {
+	return apiErrs.NewCustomValidationError(
+		fmt.Sprintf(
+			"requirement failed: Wrong char %q in Base58 string '%s'",
+			invalidChar,
+			id,
+		),
+	)
+}
+
+func (a *NodeApi) EthereumDAppABI(w http.ResponseWriter, r *http.Request) error {
+	s := chi.URLParam(r, "address")
+	addr, err := proto.NewAddressFromString(s)
+	if err != nil {
+		if invalidRune := findFirstInvalidRuneInBase58String(s); invalidRune != nil {
```

Comment:
Why for report the invalid Base58 rune? Is it not enough to report that the address is invalid? I think that the `findFirstInvalidRuneInBase58String` is highly non-optimal, why not just lookup for chars from Base58 alphabet?

Reply:
Because we have to support scala's errors. Ok, I'll optimize it.